src/internal/clusterstate/v2.8.0/pfsdb.go

```diff
@@ -422,20 +415,60 @@ func migratePage(ctx context.Context, tx *pachsql.Tx, page []commitCollection) e
 		if !commit.StartTime.Valid {
 			return errors.Errorf("commit %s has a nil start time", commit.IntID)
 		}
-		_, err = tx.NamedExecContext(ctx, migrateQuery, commit)
+		err = batcher.Add(migrateCommit(commit))
 		if err != nil {
-			return errors.Wrap(err, fmt.Sprintf("migrating commit %d", commit.IntID))
+			return errors.Wrap(err, "batching commit migration query")
 		}
 		if ancestry.ParentCommit == "" && ancestry.ChildCommits == nil {
 			continue
 		}
-		if err := migrateRelatives(ctx, tx, commit.IntID, ancestry); err != nil {
-			return errors.Wrap(err, fmt.Sprintf("migrating relatives of commit %d", commit.IntID))
+		err = batcher.Add(migrateRelatives(commit.IntID, ancestry))
+		if err != nil {
+			return errors.Wrap(err, "batching commit ancestry query")
 		}
 	}
 	return nil
 }
 
+func migrateCommit(commit *Commit) string {
+	migrateQuery := `WITH repo_row_id AS (SELECT id from pfs.repos WHERE name='%s' AND type='%s' 
+                                                AND project_id=(SELECT id from core.projects WHERE name='%s'))
+		UPDATE pfs.commits SET 
+			commit_id='%s', 
+			commit_set_id='%s',
+		    repo_id=(SELECT id from repo_row_id), 
+		    branch_id=(SELECT id from pfs.branches WHERE name='%s' AND repo_id=(SELECT id from repo_row_id)), 
+			description='%s', 
+			origin='%s', 
+			start_time='%v', 
+			%s 
+			%s
+			%s
+			%s
+			size=%v, 
+			error='%s' 
+		WHERE int_id=%d;`
+	finishingTime := ""
+	finishedTime := ""
+	compactingTime := ""
+	validatingTime := ""
+	if commit.FinishedTime.Valid {
+		finishedTime = fmt.Sprintf("finished_time='%v',", commit.FinishedTime.Time.Format(time.RFC3339))
+	}
+	if commit.FinishedTime.Valid {
+		finishingTime = fmt.Sprintf("finishing_time='%v',", commit.FinishingTime.Time.Format(time.RFC3339))
+	}
+	if commit.CompactingTime.Valid {
+		compactingTime = fmt.Sprintf("compacting_time_s=%v,", commit.CompactingTime.Int64)
+	}
+	if commit.ValidatingTime.Valid {
+		validatingTime = fmt.Sprintf("validating_time_s=%v,", commit.ValidatingTime.Int64)
+	}
+	return fmt.Sprintf(migrateQuery, commit.RepoName, commit.RepoType, commit.ProjectName, commit.CommitID, commit.CommitSetID,
+		commit.BranchName.String, commit.Description, commit.Origin, commit.StartTime.Time.Format(time.RFC3339), finishingTime, finishedTime,
```

Comment:
Most of these fields have a restricted alphabet (like the branch name).  But commit description does not.
https://xkcd.com/327/